### ISSUE TITLE
`LLMModel` does not support tool calling

### ISSUE THREAD

**Issue by jamesbraza** [Future-House/paper-qa#701]
It would be nice if `LLMModel` directly supported tool calling.

That way, we wouldn't need to shim to `aviary.ToolSelector` like we do here: https://github.com/Future-House/paper-qa/blob/6fda922f345a5d9cd38e5a6bce6de7a055e6f44d/paperqa/llms.py#L757-L764

**Comment by maykcaldas** [comment]
I think this is done, but select_tool is supported by `LiteLLMModel`: https://github.com/Future-House/ldp/blob/main/packages/lmi/src/lmi/llms.py#L708

`select_tool` is now used from the `LLM`: https://github.com/Future-House/paper-qa/blob/ca48144da75bb6ef217af7c24f4aa48f2c4814e1/paperqa/agents/main.py#L248

Closing this issue for now, as it seems to be solved. Feel free to reopen it

**Comment by jamesbraza** [comment]
Yeah it looks like this was ported to `llm-client` in https://github.com/Future-House/llm-client/pull/2

Thanks for looking into this!
